wren-ui/src/apollo/server/services/askingService.ts

```diff
@@ -581,31 +745,103 @@ export class AskingService implements IAskingService {
         throw new Error(`View ${input.viewId} not found`);
       }
 
-      const res = await this.createThreadResponseFromView(input, view, thread);
+      const res = await this.createThreadResponseFromView(
+        input.question,
+        view.statement,
+        view,
+        thread,
+      );
       return res;
     }
 
+    const threadResponse = await this.threadResponseRepository.createOne({
+      threadId: thread.id,
+      question: input.question,
+      sql: input.sql,
+    });
+
+    return threadResponse;
+  }
+
+  public async generateThreadResponseBreakdown(
+    threadId: number,
+    threadResponseId: number,
+    configurations: { language: string },
+  ): Promise<ThreadResponse> {
+    const { language } = configurations;
+    const thread = await this.threadRepository.findOneBy({
+      id: threadId,
+    });
+    const threadResponse = await this.threadResponseRepository.findOneBy({
+      id: threadResponseId,
+    });
+
+    if (!thread) {
```

Comment:
I think checking the threadResponse is enough here. 
I found @andreashimin use the same pattern in the text-to-chart PR and I ask [the same question](https://github.com/Canner/WrenAI/pull/997#discussion_r1886921822) there.

Reply:
resolved in [fa383c9](https://github.com/Canner/WrenAI/pull/986/commits/fa383c92c402135cfcd9196a0bf638966dd7ca2d)